### DISCUSSION TITLE
Git push failed

### DISCUSSION THREAD

**Discussion by ChristianSoh72** [community/community#63670] [New to GitHub]
### Body

I've developed a project, and then I wanted to upload it on github.
So I created repository on github, and then set my  project's remote origin as this repository.

After committing all the changes, I've pushed.
But it says error message can't push "fatal: refusing to merge unrelated histories".
Why this comes?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by nirav-gajera** [comment]
To fix this error, you need to either merge the histories of the two repositories, or you need to reset your local repository to the state of the remote repository.

To merge the histories of the two repositories, you can use the git merge command. For example, if the name of your remote repository is `origin`, you would use the following command:

```
git merge origin/master
```

This command will merge the `master` branch of the remote repository into your local repository. If there are any conflicts, you will need to resolve them manually.

To reset your local repository to the state of the remote repository, you can use the git reset command. For example, if the name of your remote repository is `origin`, you would use the following command:

```
git reset --hard origin/master
```
